src/asm/lexer.c

```diff
@@ -2267,11 +2269,13 @@ static int yylex_SKIP_TO_ENDR(void)
 
 int yylex(void)
 {
-restart:
-	if (lexerState->atLineStart && lexerStateEOL) {
+	if (lexerStateEOL) {
```

Comment:
Why was the `lineStart` check removed? If it's not needed anymore because the mode switches always occur at EOL, then they should be made instant outright, instead of keeping the deferring mechanism.

Reply:
The tests pass whether or not `lexerState->atLineStart &&` is there, but doing `lexer_SetState(lexerStateEOL); lexerStateEOL = NULL;` right after reaching EOB causes the `assert(state != lexerStateEOL)` in `lexer_DeleteState` to fail. I'm just putting `lexerState->atLineStart &&` back.

Reply:
This needs further checking.